libmisc/salt.c

```diff
@@ -125,23 +124,27 @@ static /*@observer@*/char *l64a (long value)
 }
 #endif /* !HAVE_L64A */
 
-static void seedRNG (void)
+/* Read sizeof (long) random bytes from /dev/urandom. */
+static long read_random_bytes (void)
 {
-	struct timeval tv;
-	static int seeded = 0;
+	long randval = 0;
+	FILE *f = fopen ("/dev/urandom", "r");
```

Comment:
I fear we need to support systems without urandom.  Ideally read_random_bytes() would read urandom if possible, and fall back if not.  I realize that may open the door to certain attacks, but an unprivileged user shouldn't be able to hide /dev/urandom, at least.

Am I wrong?  I'm actually not sure what the full list of platforms using this is.

Reply:
`/dev/urandom` by the time of writing is well supported by Linux, Solaris, {Free,Net,Open}BSD, Tru64 UNIX 5.1B, AIX 5.2, and HP-UX 11i v2.  Cygwin emulates it.  I don't think relying on its availability would be a blocker here.

Reply:
Ok - I'm pretty sure you're right.  let's see if anyone objects at next release :)

Thanks